src/edu/csus/ecs/pc2/clics/API202306/CLICSEndpoint.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS Endpoint
+ * Contains information about an API endpoint that is supported
+ * 
+ * @author John Buck
+ *
+ */
+
+public class CLICSEndpoint {
+
+    @JsonProperty
+    private String type;
+
+    @JsonProperty
+    private String [] properties;
+
+    /**
+     * For use with the access endpoint.  This describes the properties of a single endpoint.
+     * 
+     * @param type String representing the name of the endpoint, eg. "teams", "groups", etc.
+     * @param properties List of supported properties
+     */
+    public CLICSEndpoint(String type, String [] properties) {
+        this.type = type;
+        this.properties = properties;
```

Comment:
The CLICS spec says `the array [of supported properties] must not be empty.`  This seems to imply that the constructor shouldn't allow (that is, should check for) an empty or null `properties` parameter.  (Either that or we need to ensure that every call to this constructor enforces this condition -- which doesn't seem viable and certainly doesn't protect against "future code").   Not sure what to do here -- seems like "throw an exception" is right, but what exception and who will catch it?

Reply:
Well, I don't see it being an issue because it would imply that a CLICSxxxxx class had NO properties (this constructor is only used by the `getEndpointProperties `method in each service.  I don't think there are any, or will be any classes that have no properties, and, if they do, we have bigger problems than a spec violation.